### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by wilwade**
Updated _almost_ all of the dependencies to the latest version.

- Bson updates required a few small changes due to es module updates in bson
- Did *not* update chai past v4. Looks like moving past chai v4 would require many changes and not all the chai plugins we are using support it yet. Perhaps we should just move away from chai instead.
- Moved from ts-node to tsx
- Updates for `msw` breaking changes, but nothing special for our use cases.
- Github actions with node no longer needs a separate cache step

## Suggested Tests
- Locally build and run tests
- Try it out with:
    - `npm i && npm run build && npm run serve`
    - `cd examples/server && npm i && node app.js`
    - Then go to `http://localhost:3000`, open up the console and hit some buttons.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -6,22 +6,15 @@ on:
...
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Use Node.js
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v4
         with:
-          node-version: '18.18.2'
-
-      - uses: actions/cache@v2
```
Comment (wilwade) [inline .github/workflows/main.yml:19]: `actions/setup-node@v4` has this included

File: .github/workflows/publish-next.yml [hunk 2]
```diff
@@ -4,24 +4,17 @@ on:
...
 
       - name: Use Node.js
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v4
         with:
-          node-version: '18.18.2'
+          node-version: 18
           registry-url: 'https://registry.npmjs.org'
 
-      - uses: actions/cache@v2
```
Comment (wilwade) [inline .github/workflows/publish-next.yml:18]: `actions/setup-node@v4` has this included

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -4,24 +4,17 @@ on:
...
 
       - name: Use Node.js
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v4
         with:
-          node-version: '18.18.2'
+          node-version: 18
           registry-url: 'https://registry.npmjs.org'
 
-      - uses: actions/cache@v2
```
Comment (wilwade) [inline .github/workflows/release.yml:18]: `actions/setup-node@v4` has this included

File: gen-nodejs/parquet_types.d.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 //
-// Autogenerated by Thrift Compiler (0.16.0)
+// Autogenerated by Thrift Compiler (0.18.1)
```
Comment (wilwade) [inline gen-nodejs/parquet_types.d.ts:2]: I did regen these files with v0.18.1, but nothing was different except this header.

File: lib/types.ts [hunk 5]
```diff
@@ -1,8 +1,11 @@
 'use strict';
 // Thanks to https://github.com/kbajalc/parquets for some of the code.
-import * as BSON from "bson"
-import { PrimitiveType, OriginalType, ParquetType, FieldDefinition, ParquetField } from "./declare"
+import { PrimitiveType, OriginalType, ParquetType, FieldDefinition, ParquetField } from "./declare";
 import { Options } from "./codec/types";
+import type { Document as BsonDocument } from "bson";
+// BSON uses top level awaits, so use require for now
+const bsonSerialize = require('bson').serialize;
+const bsonDeserialize = require('bson').deserialize;
```
Comment (wilwade) [inline lib/types.ts:8]: Really annoying and only impacts esbuild for the browser, but not too bad.

Why the double require? It was clearer than the named version and keeps the linter away from trying to reformat it.

File: test/mocks/handlers.js
Comment (wilwade) [inline test/mocks/handlers.js:1]: These were updates for the new version of `msw`. No big changes however

File: tsconfig.json [hunk 6]
```diff
@@ -12,15 +12,11 @@
     "sourceMap": false,
     "strict": true,
     "target": "ESNext",
-    "watch": false,
```
Comment (wilwade) [inline tsconfig.json:15]: This never did anything.

File: tsconfig.json [hunk 7]
```diff
@@ -12,15 +12,11 @@
...
     "typeRoots": [
       "node_modules/@types",
       "gen-nodejs",
     ]
   },
   "include":  ["parquet.ts","lib/**/*", "gen-nodejs/*", "browser/*.ts"],
-  "files": ["lib/custom.d.ts"],
-  "ts-node": {
-    "files": true
-  }
```
Comment (wilwade) [inline tsconfig.json:25]: No more ts-node

**Review by shannonwells — COMMENTED** [review]
I got this error on `cd examples && npm i && node app.js`:
```shell
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/Users/shannonwells/github/parquetjs/examples/app.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:86:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

File: lib/types.ts [hunk 8]
```diff
@@ -1,8 +1,11 @@
 'use strict';
 // Thanks to https://github.com/kbajalc/parquets for some of the code.
-import * as BSON from "bson"
-import { PrimitiveType, OriginalType, ParquetType, FieldDefinition, ParquetField } from "./declare"
+import { PrimitiveType, OriginalType, ParquetType, FieldDefinition, ParquetField } from "./declare";
 import { Options } from "./codec/types";
+import type { Document as BsonDocument } from "bson";
+// BSON uses top level awaits, so use require for now
+const bsonSerialize = require('bson').serialize;
```
Comment (shannonwells) [inline lib/types.ts:7]: 🙄 THANKS BSON

**Review by shannonwells — APPROVED** [review]
I checked out the branch and ran it and tested the example "app".
I reviewed the code.
I was disappointed that the simple 'rest' interface, which I guess is gone/deprecated, had to be replaced by http with string keys.

**Pull request merged**